ruby_on_rails/apis/apis_and_building_your_own.md

```diff
@@ -192,30 +191,32 @@ In the next lesson, we'll cover working with other people's APIs, which can add
-- <a class="knowledge-check-link" href="#rendering-json-or-xml">What is the purpose of the `#respond_to` method?</a>
-- <a class="knowledge-check-link" href="#specifying-attributes-to-return">How do you return a User object but specify that you don't want to include certain attributes (i.e. you can't just return `User.first`)?</a>
-- <a class="knowledge-check-link" href="#to-json-steps">What are the two steps performed behind the scenes by the `#to_json` method?</a>
-- <a class="knowledge-check-link" href="https://guides.rubyonrails.org/layouts_and_rendering.html#using-head-to-build-header-only-responses">How do you tell a controller action to render nothing but an error message?</a>
-- <a class="knowledge-check-link" href="https://web-crunch.com/posts/custom-error-page-ruby-on-rails">How do you build your own custom error messages?</a>
-- <a class="knowledge-check-link" href="#api-tokens">Why can't you use session-based controller authentication methods if you want people to access your API programmatically?</a>
-- <a class="knowledge-check-link" href="#service-oriented-architecture-soa">What is "Service Oriented Architecture?</a>
+- [How does Rails know which type of file you are expecting back when you make an HTTP request?](#http-request-format)
+- [What is the purpose of the `#respond_to` method?](#rendering-json-or-xml)
+- [How do you return a User object but specify that you don't want to include certain attributes (i.e. you can't just return `User.first`)?](#specifying-attributes-to-return)
+- [What are the two steps performed behind the scenes by the `#to_json` method?](#to-json-steps)
+- [How do you tell a controller action to render nothing but an error message?](https://guides.rubyonrails.org/layouts_and_rendering.html#using-head-to-build-header-only-responses)
+- [How do you build your own custom error messages?](https://web-crunch.com/posts/custom-error-page-ruby-on-rails)
+- [Why can't you use session-based controller authentication methods if you want people to access your API programmatically?](#api-tokens)
+- [What is "Service Oriented Architecture?](#service-oriented-architecture-soa)
 
 ### Additional resources
 
-This section contains helpful links to related content. It isn’t required, so consider it supplemental.
+This section contains helpful links to related content. It isn't required, so consider it supplemental.
 
-- Watch [this free Railscast on making your App into an API](http://railscasts.com/episodes/348-the-rails-api-gem)
-- Watch [this free Railscast on securing your API](http://railscasts.com/episodes/352-securing-an-api)
-- Watch [this free Railscast on versioning your API](http://railscasts.com/episodes/350-rest-api-versioning)
+- Watch this free [Railscast on making your App into an API](http://railscasts.com/episodes/352-securing-an-api)
+- Watch this free [Railscast on versioning your API](http://railscasts.com/episodes/352-securing-an-api)
```

Comment:
Looks like the same link got repeated here twice and the description got off on the second one.

Reply:
That'll be my fault. Botched the copy+paste for a few cells in the spreadsheet (which have now been fixed :eyes:)

Reply:
Replaced them 👍🏻